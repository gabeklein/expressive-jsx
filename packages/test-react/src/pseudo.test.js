const test = require("./_adapter");

describe("pseudo", () => {
  test("before/after", `
    () => {
      css: before: {
        content: "This is content"
        color: blue;
      }

      css: after: {
        background: red;
        color: white;
      }

      <div>Hello</div>
    }
  `);

  test("nthOfType", `
    () => {
      css: nthOfType: {
        select: "2n"
        color: blue;
      }
      
      css: nthOfType: {
        select: "3n+2"
        color: red;
      }

      <div>Hello</div>
    }
  `);
});

describe("css", () => {
  test("directive", `
    () => {
      color: blue;
      
      css: self: {
        isRed: {
          color: red;
        }
      }

      <div>Hello</div>
    }
  `);

  test("multiple directives", `
    () => {
      color: blue;
      
      css: self: {
        isRed: {
          color: red;
        }

        isGreen: {
          color: green;
        }
      }

      <div>Hello</div>
    }
  `);
});