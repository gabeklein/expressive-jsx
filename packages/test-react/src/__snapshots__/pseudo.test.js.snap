// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css directive 1`] = `
"() => create(\\"div\\", {
  className: \\"element_7nq\\"
}, create(\\"div\\", {}, \\"Hello\\"));

css(\`
	/* 1 */
	.element_7nq { color: blue }
	/* 5 */
	.element_7nq.isRed { color: red }
\`);"
`;

exports[`css directive 2`] = `
"() => <div className=\\"element_7nq\\"><div>Hello</div></div>;

css(\`
	/* 1 */
	.element_7nq { color: blue }
	/* 5 */
	.element_7nq.isRed { color: red }
\`);"
`;

exports[`css multiple directives 1`] = `
"() => create(\\"div\\", {
  className: \\"element_7nq\\"
}, create(\\"div\\", {}, \\"Hello\\"));

css(\`
	/* 1 */
	.element_7nq { color: blue }
	/* 5 */
	.element_7nq.isRed { color: red }
	.element_7nq.isGreen { color: green }
\`);"
`;

exports[`css multiple directives 2`] = `
"() => <div className=\\"element_7nq\\"><div>Hello</div></div>;

css(\`
	/* 1 */
	.element_7nq { color: blue }
	/* 5 */
	.element_7nq.isRed { color: red }
	.element_7nq.isGreen { color: green }
\`);"
`;

exports[`pseudo before/after 1`] = `
"() => create(\\"div\\", {
  className: \\"element_7nq\\"
}, create(\\"div\\", {}, \\"Hello\\"));

css(\`
	/* 6 */
	.element_7nq::before { content: \\"This is content\\"; color: blue }
	.element_7nq::after { background: red; color: white; content: \\"\\" }
\`);"
`;

exports[`pseudo before/after 2`] = `
"() => <div className=\\"element_7nq\\"><div>Hello</div></div>;

css(\`
	/* 6 */
	.element_7nq::before { content: \\"This is content\\"; color: blue }
	.element_7nq::after { background: red; color: white; content: \\"\\" }
\`);"
`;

exports[`pseudo nthOfType 1`] = `
"() => create(\\"div\\", {
  className: \\"element_7nq\\"
}, create(\\"div\\", {}, \\"Hello\\"));

css(\`
	/* 6 */
	.element_7nq:nth-of-type(2n) { color: blue }
	.element_7nq:nth-of-type(3n+2) { color: red }
\`);"
`;

exports[`pseudo nthOfType 2`] = `
"() => <div className=\\"element_7nq\\"><div>Hello</div></div>;

css(\`
	/* 6 */
	.element_7nq:nth-of-type(2n) { color: blue }
	.element_7nq:nth-of-type(3n+2) { color: red }
\`);"
`;
